Tests/DeepLearningTests/LayerTests.swift

```diff
@@ -31,15 +31,47 @@ final class LayerTests: XCTestCase {
         let input = Tensor<Float>([[0, 1, 2, 3, 4], [10, 11, 12, 13, 14]]).expandingShape(at: 2)
         let output = layer.inferring(from: input)
         let expected = Tensor<Float>([[[2], [3], [4]], [[12], [13], [14]]])
-        XCTAssertEqual(round(output), expected)
+        XCTAssertEqual(output, expected)
+    }
+
+    func testMaxPool2D() {
+        let layer = MaxPool2D<Float>(poolSize: (2, 2), strides: (1, 1), padding:.valid)
+        let input = Tensor(shape: [1, 2, 2, 1], scalars: (0..<4).map(Float.init))
+        let output = layer.inferring(from: input)
+        let expected = Tensor<Float>([[[[3]]]])
+        XCTAssertEqual(output, expected)
+    }
+
+    func testMaxPool3D() {
+        let layer = MaxPool3D<Float>(poolSize: (2 ,2, 2), strides: (1, 1, 1), padding:.valid)
```

Comment:
Minor formatting: space after comma (instead of ahead comma), space after colon.
```suggestion
        let layer = MaxPool3D<Float>(poolSize: (2, 2, 2), strides: (1, 1, 1), padding:.valid)
```

I'll go ahead and push edits for you.

Reply:
@dan-zheng the errors i mentioned in the last few messages, could you tell me why they popped up? I'd like to understand that.

Reply:
Oops, sorry I overlooked that error - I should've tested first.
I'll investigate now and cc you on a fix PR.